File: src/main/scala/com/blackwing/easyExploration/blocks/Bases/BlockBase.scala
package com.blackwing.easyExploration.blocks.Bases

import com.blackwing.easyExploration.EasyExploration
import com.blackwing.easyExploration.utilities.IHasModel
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraft.creativetab.CreativeTabs
import net.minecraft.item.{Item, ItemBlock}

class BlockBase(name: String, material: Material) extends Block(material) with IHasModel {

	val tab: CreativeTabs = CreativeTabs.BUILDING_BLOCKS

	setUnlocalizedName(name)
	setRegistryName(name)
	setCreativeTab(tab)
	EasyExploration.blocks :+= this
	EasyExploration.items :+= new ItemBlock(this).setRegistryName(this.getRegistryName)

	def registerModels(): Unit = EasyExploration.proxy.registerItemRenderer(Item.getItemFromBlock(this), 0, "inventory")
}

File: src/main/java/com/blackwing/easyExploration/block/base/BlockBase.java
package com.blackwing.easyExploration.block.base;

import net.minecraft.block.Block;
import net.minecraft.block.material.Material;

public class BlockBase extends Block implements IBlock {

    private String id;

    public void setId(String id) {
        this.id = id;
    }

    public String getId() {
        return id;
    }

    public BlockBase(String id, Material material) {
        super(material);
        init(id);
    }
}
